cli/src/config.rs

```diff
@@ -231,18 +245,26 @@ impl UnresolvedConfigEnv {
             config_dir.push("conf.d");
             ConfigPath::new(config_dir)
         });
-        use ConfigPath::*;
-        if let Some(path @ Existing(_)) = home_config_path {
-            paths.push(path);
-        } else if let (Some(path @ New(_)), None) = (home_config_path, &platform_config_path) {
-            paths.push(path);
+
+        if let Some(path) = home_config_path {
+            if path.exists() || platform_config_path.is_none() {
+                paths.push(path);
+            }
         }
+
         // This should be the default config created if there's
         // no user config and `jj config edit` is executed.
         if let Some(path) = platform_config_path {
             paths.push(path);
         }
-        if let Some(path @ Existing(_)) = platform_config_dir {
+
+        // theoretically this should be an `if let Some(...) = ... && ..., but that
+        // isn't stable
+        let get_existing = |opt: Option<ConfigPath>| match opt {
+            Some(path) if path.exists() => Some(path),
+            _ => None,
+        };
```

Comment:
nit: this helper function appears to be deleted in the next commit. Just inline it?

Reply:
oops, seems like we had a rebase issue